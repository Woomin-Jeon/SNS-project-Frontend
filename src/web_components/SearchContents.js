import { Link } from 'react-router-dom';
import React from 'react';

function SearchContents({
  searchState,
  setSearchState,
  topLevelState,
  setTopLevelState,
  loginState,
}) {
  const { users } = loginState;

  const setUserToTopLevelByID = (id) => {
    for (let i = 0; i < users.length; i += 1) {
      if (id === users[i].id) {
        setTopLevelState({
          ...topLevelState,
          id: users[i].id,
          name: users[i].userName,
        });
      }
    }
    setSearchState({ ...setSearchState, contents: [], exist: false });
  };

  return (
    <div className="join-following">
      {searchState.exist
        ? (
          searchState.contents.map((v, index) => (
            <div key={index}>
              <label className="join-users">
                <img className="join-user-profile" src={v.profile} alt="" />
                <div className="join-users-informatrion">
                  <Link
                    to="/otherspage"
                    className="join-users-name"
                    type="button"
                    onClick={() => setUserToTopLevelByID(v.id)}
                  >
                    {v.userName}
                  </Link>
                  <div className="join-users-lint">{v.birth}</div>
                  <div className="join-users-lint">{v.location}</div>
                  <div className="join-users-lint">{v.email}</div>
                </div>
              </label>
            </div>
          )))
        : (<span>해당 유저가 존재하지 않습니다</span>)}
    </div>
  );
}

export default SearchContents;
