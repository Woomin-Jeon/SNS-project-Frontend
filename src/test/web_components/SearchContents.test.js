import SearchContents from "../../web_components/SearchContents";
import React from "react";
import {render} from 'enzyme';

describe('SearchContents', () => {
  let searchState;
  let setSearchState;
  let topLevelState;
  let setTopLevelState;
  let loginState;

  beforeEach(() => {
    searchState = {
      exist: false,
      contents: [],
    };
    loginState = {
      users: [
        {
          id: 'Random_ID',
          profile: 'path',
          userName: 'Woomin',
        }
      ]
    };
  });

  it('renders well', () => {
    const component = render(
      <SearchContents
        searchState={searchState}
        setSearchState={setSearchState}
        topLevelState={topLevelState}
        setTopLevelState={setTopLevelState}
        loginState={loginState}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toMatch('해당 유저가 존재하지 않습니다');
  });
});
