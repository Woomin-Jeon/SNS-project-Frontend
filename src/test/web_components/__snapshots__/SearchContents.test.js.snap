// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContents renders well 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "join-following",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "해당 유저가 존재하지 않습니다",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
